### DISCUSSION TITLE
Fundamentos C# Variáveis

### DISCUSSION THREAD

**Discussion by fredcdf** [balta-io/forum#229] [Tira Dúvidas] [answered]
Tem algum problema em usar variáveis em português com acentuação gráfica?

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @fredcdf, como vai?
Não tem problema ser em português, porém acentos, espaços e caracteres especiais não devem ser utilizados.

**Reply by fredcdf** [reply]
espaços, caracteres especiais não devem ser utilizados, mas qual o motivo? gostaria de entender, pois hoje todos editores são unicode, etc... 
Em especial qual problema em usar palavras em português com acentos? Pois tenho feito teste com acentuação em portugês e não tenho tido problema. Gostaria de entender onde está o problema de fato. Com padrão antigo ASC etc... não tinha como mas hoje mudou muito isso com unicode por exemplo. Em que momento que meus aplicativos com acentos darão problema?

Há uma diferença grande na interpretação: "Não devem ser usados" pode ser "não podem" ou pode ser "não indicado apenas"  Caso não indicado na interpretação: não dando erro na compilação e nem na execução onde estaria de fato o problema ... gostaria que fosse mais claro. Dúvida boba mas estou curioso pois gosto de codificar em Português apesar de não ser leigo em inglês!

**Reply by andrebaltieri** [reply]
Bom dia, @fredcdf, como vai?
Unicode é um padrão que define um valor numérico para um caractere. UTF é o tipo de codificação que transforma este código em um símbolo.

Embora utilize unicode, há uma garantia maior quando usamos apenas os caracteres do BMP (Basic Model Plan), que no caso do UTF 8 incluem de A a Z (Maiúsculo e minúsculo) e números. Como o UTF-8 também é compatível com ASCII, fica mais tranquilo ainda.

Quando você tem um acento, são usados dois code points pra mapear este caractere, isto é conhecido como surrogate pair (par substitutivo), e quando você tem um emoji por exemplo, são utilizados até quatro pontos (graphene cluster).

Então não é proibido usar acentos, caracteres especiais e até emojis, mas sim uma boa prática. Mantendo o padrão básico, os caracteres que compoe a raiz (ASCII), garantimos que tudo fica 100%, independente do UTF.
